RecoParticleFlow/PFProducer/interface/PFTables.h

```diff
@@ -0,0 +1,194 @@
+  std::unordered_map<size_t, std::set<size_t>> cluster_to_rechit_;
+
+  PFClusterTables() = default;
+  PFClusterTables(ClusterTable&& _cluster_table,
+                  RecHitTable&& _rechit_table,
+                  std::vector<size_t>&& _cluster_to_element,
+                  std::vector<size_t>&& _element_to_cluster,
+                  std::unordered_map<size_t, std::set<size_t>>&& _rechit_to_cluster,
+                  std::unordered_map<size_t, std::set<size_t>>&& _cluster_to_rechit)
+      : cluster_table_{_cluster_table},
+        rechit_table_{_rechit_table},
+        cluster_to_element_{_cluster_to_element},
+        element_to_cluster_{_element_to_cluster},
+        rechit_to_cluster_{_rechit_to_cluster},
+        cluster_to_rechit_{_cluster_to_rechit} {};
+
+  void clear() {
+    cluster_table_.resize(0);
+    rechit_table_.resize(0);
+    cluster_to_element_.clear();
+    element_to_cluster_.clear();
+    rechit_to_cluster_.clear();
+    cluster_to_rechit_.clear();
+  }
+};
+
+class PFTables {
+public:
+  std::vector<size_t> track_to_element_;
+  std::vector<size_t> element_to_track_;
+  std::vector<std::vector<size_t>> track_to_convrefs_;
+
+  edm::soa::ConvRefTable convref_table_;
+  edm::soa::TrackTableVertex track_table_vertex_;
+  edm::soa::TrackTableExtrapolation track_table_ecalshowermax_;
+  edm::soa::TrackTableExtrapolation track_table_hcalent_;
+  edm::soa::TrackTableExtrapolation track_table_hcalex_;
+  edm::soa::TrackTableExtrapolation track_table_vfcalent_;
+  edm::soa::TrackTableExtrapolation track_table_ho_;
+
+  edm::soa::GSFTable gsf_table_;
+  edm::soa::ConvBremTable gsf_convbrem_table_;
+  edm::soa::TrackTableExtrapolation gsf_table_ecalshowermax_;
+  edm::soa::TrackTableExtrapolation gsf_table_hcalent_;
+  edm::soa::TrackTableExtrapolation gsf_table_hcalex_;
+
+  std::vector<size_t> gsf_to_element_;
+  std::vector<size_t> element_to_gsf_;
+  std::vector<std::vector<size_t>> gsf_to_convbrem_;
+
+  std::vector<size_t> element_to_brem_;
+  edm::soa::BremTable brem_table_;
+  edm::soa::TrackTableExtrapolation brem_table_ecalshowermax_;
+  edm::soa::TrackTableExtrapolation brem_table_hcalent_;
+
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_ps1_;
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_ps2_;
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_hcal_;
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_ecal_;
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_ho_;
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_hfem_;
+  PFClusterTables<edm::soa::ClusterTable, edm::soa::RecHitTable> clusters_hfhad_;
+  PFClusterTables<edm::soa::SuperClusterTable, edm::soa::SuperClusterRecHitTable> clusters_sc_;
+
+  // edm::soa::SuperClusterTable superclusters_table_;
+  // std::vector<size_t> element_to_supercluster_;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant